### ISSUE TITLE
git pile genlinear-branch fails

### ISSUE THREAD

**Issue by ashutoshx** [git-pile/git-pile#90]
```
[igt-gpu-tools]$ git pile genlinear-branch -b internal-linear
Traceback (most recent call last):
  File "/home/adixit/.local/bin/git-pile", line 4, in <module>
    __import__('pkg_resources').run_script('git-pile==1.0.dev0', 'git-pile')
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 656, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1453, in run_script
    exec(code, namespace, namespace)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/EGG-INFO/scripts/git-pile", line 12, in <module>
    sys.exit(git_pile.main(*sys.argv[1:]))
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 2432, in main
    return args.func(args, config)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 1698, in cmd_genlinear_branch
    parent_ref, pile_range = get_refs_from_linearized(args.incremental, config.pile_branch, args.start_ref, branch, notes_ref)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 1660, in get_refs_from_linearized
    return None, pile_range
UnboundLocalError: local variable 'pile_range' referenced before assignment
```

**Comment by ashutoshx** [comment]
git pile genbranch also seems busted. ec70945 seems to be the last working commit. Cc: @lucasdemarchi, @guludo 
```
git pile genbranch -f -i
error: unknown option `format=%(objectname) %(objecttype) %(path)'
usage: git ls-tree [<options>] <tree-ish> [<path>...]

    -d                    only show trees
    -r                    recurse into subtrees
    -t                    show trees when recursing
    -z                    terminate entries with NUL byte
    -l, --long            include object size
    --name-only           list only filenames
    --name-status         list only filenames
    --full-name           use full path names
    --full-tree           list entire tree; not just current directory (implies --full-name)
    --abbrev[=<n>]        use <n> digits to display object names

Traceback (most recent call last):
  File "/home/adixit/.local/bin/git-pile", line 4, in <module>
    __import__('pkg_resources').run_script('git-pile==1.0.dev0', 'git-pile')
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 656, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1453, in run_script
    exec(code, namespace, namespace)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/EGG-INFO/scripts/git-pile", line 12, in <module>
    sys.exit(git_pile.main(*sys.argv[1:]))
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 2432, in main
    return args.func(args, config)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 1645, in cmd_genbranch
    return _genbranch(root, patchesdir, config, args)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/git_pile.py", line 1542, in _genbranch
    effective_baseline, patchlist_offset = cache.search_best_base(pile_for_cache)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/genbranch_caching.py", line 81, in search_best_base
    baseline = pile.baseline()
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/pile.py", line 119, in baseline
    return self.__get_config().get("BASELINE")
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/pile.py", line 150, in __get_config
    for i, line in enumerate(self.__reader.text("config").splitlines()):
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/pile.py", line 288, in text
    self.__ls_path(path)
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/pile.py", line 301, in __ls_path
    info = self.__ls_tree()
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/pile.py", line 326, in __ls_tree
    for entry in _git(cmd).stdout.rstrip("\x00").split("\x00"):
  File "/home/adixit/.local/lib/python3.10/site-packages/git_pile-1.0.dev0-py3.10.egg/git_pile/helpers.py", line 94, in __call__
    ret = subprocess.run(l, *args, **kwargs)
  File "/usr/lib64/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'ls-tree', '-z', '--full-tree', '--format=%(objectname) %(objecttype) %(path)', 'b544f552cc0afad63632fdfd3cf74dd4cf2c7d3e']' returned non-zero exit status 129.
```

**Comment by lucasdemarchi** [comment]
What's your git version?

**Comment by ashutoshx** [comment]
2.34.1

**Comment by lucasdemarchi** [comment]
@guludo --format argument to ls-tree was only added in git 2.36. I guess that is too new to be a requirement, so we may need to fallback to something else.

**Comment by guludo** [comment]
The fix for genbranch is at #91.

I think the error for genlinear-branch is unrelated though.
